glob_test.go

```diff
@@ -450,6 +450,25 @@ func TestGlob(t *testing.T) { // nolint:funlen
 			"a/b/4.txt",
 		}, matches)
 	})
+
+	t.Run("symlinks", func(t *testing.T) {
+		t.Parallel()
+		testFS := testFs(t, []string{
+			"./a/file",
+		}, nil).(testfs.FS)
+
+		fsPath := testFS.Path()
+		os.Symlink("file", filepath.Join(fsPath, "./a/working-symlink"))
+		os.Symlink("non-existent", filepath.Join(fsPath, "./a/broken-symlink"))
+
+		matches, err := Glob("a/*", MatchDirectoryIncludesContents, WithFs(testFS))
+		require.NoError(t, err)
+		require.Equal(t, []string{
+			"a/broken-symlink",
+			"a/file",
+			"a/working-symlink",
```

Comment:
if I understand it correctly, without the patch, fileglob would follow the symlink and this entry wouldn't be here, as `a/file` is already, correct?

Reply:
Please, take another look. The fix is for patterns of symlinks, despite if they are valid of point to non-existent files